certbot/certbot/_internal/cli/helpful.py

```diff
@@ -246,51 +265,51 @@ def set_test_server(self, parsed_args: argparse.Namespace) -> None:
 
         default_servers = (flag_default("server"), constants.STAGING_URI)
 
-        if parsed_args.staging and parsed_args.server not in default_servers:
+        if config.staging and config.server not in default_servers:
             raise errors.Error("--server value conflicts with --staging")
 
-        if parsed_args.server in default_servers:
-            parsed_args.server = constants.STAGING_URI
+        if config.server == flag_default("server"):
```

Comment:
`default_servers` is just two items, `flag_default("server")` and `constants.STAGING_URI`, so assigning  it `STAGING_URI` only makes sense to do if it's the `flag_default("server")` value. this would've been a nonissue before, but now that we're keeping track of runtime overrides, i felt it made sense to only change values when needed

Reply:
Ahhhh makes sense, sgtm.